src/sql/workbench/contrib/query/browser/query.contribution.ts

```diff
@@ -305,228 +305,104 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
-	'mssql.query.quotedIdentifier': {
-		'type': 'boolean',
-		'default': true,
-		'description': localize('mssql.query.quotedIdentifier', "Enable SET QUOTED_IDENTIFIER")
-	},
-	'mssql.query.ansiNullDefaultOn': {
-		'type': 'boolean',
-		'default': true,
-		'description': localize('mssql.query.ansiNullDefaultOn', "Enable SET ANSI_NULL_DFLT_ON")
-	},
-	'mssql.query.implicitTransactions': {
-		'type': 'boolean',
-		'default': false,
-		'description': localize('mssql.query.implicitTransactions', "Enable SET IMPLICIT_TRANSACTIONS")
-	},
-	'mssql.query.cursorCloseOnCommit': {
-		'type': 'boolean',
-		'default': false,
-		'description': localize('mssql.query.cursorCloseOnCommit', "Enable SET CURSOR_CLOSE_ON_COMMIT")
-	},
-	'mssql.query.ansiPadding': {
-		'type': 'boolean',
-		'default': true,
-		'description': localize('mssql.query.ansiPadding', "Enable SET ANSI_PADDING")
-	},
-	'mssql.query.ansiWarnings': {
-		'type': 'boolean',
-		'default': true,
-		'description': localize('mssql.query.ansiWarnings', "Enable SET ANSI_WARNINGS")
-	},
-	'mssql.query.ansiNulls': {
-		'type': 'boolean',
-		'default': true,
-		'description': localize('mssql.query.ansiNulls', "Enable SET ANSI_NULLS")
-	},
-	'mssql.query.alwaysEncryptedParameterization': {
-		'type': 'boolean',
-		'default': false,
-		'description': localize('mssql.query.alwaysEncryptedParameterization', "Enable Parameterization for Always Encrypted")
+const queryEditorConfiguration: IConfigurationNode = {
+	...queryEditorConfigurationBaseNode,
+	properties: {
+		'queryEditor.results.saveAsCsv.includeHeaders': {
+			'type': 'boolean',
+			'description': localize('queryEditor.results.saveAsCsv.includeHeaders', "When true, column headers are included when saving results as CSV"),
+			'default': true
+		},
+		'queryEditor.results.saveAsCsv.delimiter': {
+			'type': 'string',
+			'description': localize('queryEditor.results.saveAsCsv.delimiter', "The custom delimiter to use between values when saving as CSV"),
+			'default': ','
+		},
+		'queryEditor.results.saveAsCsv.lineSeperator': {
+			'type': 'string',
+			'description': localize('queryEditor.results.saveAsCsv.lineSeperator', "Character(s) used for seperating rows when saving results as CSV"),
+			'default': null
+		},
+		'queryEditor.results.saveAsCsv.textIdentifier': {
+			'type': 'string',
+			'description': localize('queryEditor.results.saveAsCsv.textIdentifier', "Character used for enclosing text fields when saving results as CSV"),
+			'default': '\"'
+		},
+		'queryEditor.results.saveAsCsv.encoding': {
+			'type': 'string',
+			'description': localize('queryEditor.results.saveAsCsv.encoding', "File encoding used when saving results as CSV"),
+			'default': 'utf-8'
+		},
+		'queryEditor.results.saveAsXml.formatted': {
+			'type': 'string',
+			'description': localize('queryEditor.results.saveAsXml.formatted', "When true, XML output will be formatted when saving results as XML"),
+			'default': true
+		},
+		'queryEditor.results.saveAsXml.encoding': {
```

Comment:
I changes the id of these settings. This does have a functional change in if a user has already set this setting manually, it won't pick up those settings and the user will have to reset the setting. I think we can get around this by documenting these changes. but I think its good in the long run to standardize these settings and make the naming more generic.

Reply:
What if we had a check happen at startup that looked for the old values and if they exist updated them to the new ones? We've done a similar thing in SSMS before with breaking config changes and that seems to work pretty well.

Reply:
Certainly an option.